code/__HELPERS/priority_announce.dm

```diff
@@ -44,6 +44,39 @@
 			if(M.client.prefs.toggles & SOUND_ANNOUNCEMENTS)
 				SEND_SOUND(M, s)
 
+/**
+ * Summon the crew for an emergency meeting
+ *
+ * Teleports the crew to a specified area, and tells everyone (via an announcement) who called the meeting. Should only be used during april fools!
+ * Arguments:
+ * * user - Mob who called the meeting
+ * * button_zone - Area where the meeting was called and where everyone will get teleported to
+ */
+/proc/call_emergency_meeting(mob/living/user, area/button_zone)
+	var/announcement
+
+	announcement += "<h1 class='alert'>Captain Alert</h1>"
+	//If the announcer overrides alert messages, use that message.
+	announcement += "<br><span class='alert'>[user] has called an Emergency Meeting!</span><br>"
+	announcement += "<br>"
+
+	var/s = sound('sound/misc/emergency_meeting.ogg')
+	for(var/mob/M in GLOB.mob_list) //Gotta make sure the whole crew's here!
+		if(!isnewplayer(M))
+			to_chat(M, announcement)
+			SEND_SOUND(M, s)
+			M.overlay_fullscreen("emergency_meeting", /atom/movable/screen/fullscreen/emergency_meeting, 1)
+			addtimer(CALLBACK(M, /mob/.proc/clear_fullscreen, "emergency_meeting"), 30)
+			if (is_station_level(M.z))
+				var/turf/target
+				var/list/turf_list = get_area_turfs(button_zone)
+				while (!target && turf_list.len)
+					target = pick_n_take(turf_list)
```

Comment:
Doesn't the way this while loop is set up mean that the room will fill with trash mobs and not the crew, then abort before having actually brought all the players (Who are the people we actually want to dump in a room with the captain and fifty bees)?

Reply:
nah this while loop is just for finding a suitable turf to deploy the particular mob to, the outer for-loop handles getting every single mob in there (beepsky, ian, ghosts, corpses, and the living are all valuable crewmembers!)